Adiciona Makefile para exportar org => html. Closes #18.

luksamuk opened the pull request.

luksamuk committed:
Adiciona Makefile para exportar org => html. Closes #18.

ryukinix commented:
Implementação bem safada. Digno de Makefile. Vou testar.

ryukinix commented:
![image](https://user-images.githubusercontent.com/7642878/36219214-60d182a6-1195-11e8-835a-bbce9cca8f35.png)

Existe um problema porque `--batch` assume a opção `-q` que não carrega os .init files logo não carrega o htmlize tbm :/. Logo ficamos sem highlight. Vou tentar aq criar alguns parâmetros.

ryukinix committed:
Add some control variables of makefile

Well... this should load the ~/.emacs.d/init.el which
will slow-down the compile-time but should work to enable
htmlize to do syntax highlight on source blocks.

The problem, however, seems that the theme is not working.
What I get for now is just a weird black/white syntax highlight.

There is this message as well on the end of the compilation:
Using schema /usr/share/emacs/25.3/etc/schema/xhtml.rnc

ryukinix commented:
Fiz uma pequena mudança no makefile pra carregar o `~/.emacs.d/init.el` que obviamente deixa mais lento a compilação, mas dai nem desse jeito resolveu o problema do htmlize.el, ele carrega, mas exporta parece que independente do tema... não sei se a inicialização com --batch da skip em algumas operações relacionadas com o tema... `make EMACS_FLAGS=-nw` funciona de uma maneira esquisita, mas funciona. Talvez tenha algum hook do tema que só se inicializa depois de abrir o primeiro frame...

luksamuk commented:
Será que não dá pra criar um .el que faça esse trabalho todo? Dessa forma, o `make` abre o Emacs, que carrega o .el da pasta raiz, e imediatamente executa uma função que chame o comando para escrever os HTML. Isso reduziria o tempo de compilação e provavelmente carregaria o init.el direto.

ryukinix commented:
Então, estive pensando numa coisa similar... mas de toda maneira isso significaria ter que embarcar o htmlize aqui no repositório pra poder carregar do `compile.el` (vamos dizer). Ou será que dá pra fazer alguma bruxaria dentro do .el e buscar pelo ~/.emacs.d/ local pelo htmlize e carregar? Acho que se der... deveriamos fazer. 

Talvez essa seja a solução mais simples, pq dai fica:

```
emacs <file.org> --batch --kill -l compile.el -f org-export-to-html
```

Alias, ouvi dizer que podemos compilar vários arquivos .org ao usar o --visit dentro do --batch, pq dessa maneira podemos visitar vários arquivos e compilar sem ter que abrir várias instâncias em sequencia do emacs. Isso me parece bem mais inteligente...

ryukinix commented:
Vou aceitar esse trabalho como parcialmente feito, irei dar merge, mas precisamos continuar a melhorar isso em seguida.

ryukinix reviewed (approved):
Aprovado até segunda ordem
